src/mon/OSDMonitor.cc

```diff
@@ -638,6 +638,9 @@ int OSDMonitor::reweight_by_utilization(int oload,
       // is a factor to take into account the original weights,
       // to represent e.g. differing storage capacities
       unsigned weight = osdmap.get_weight(p->first);
+      if (weight == 0) {
```

Comment:
why not move this out of this if block and right before https://github.com/ceph/ceph/pull/8941/commits/3538fff7a58e7668f6876ec398c9ac92edd25b43#diff-0a5db46a44ae9900e226289a810f10e8R634 .